controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -95,6 +114,124 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
 func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
 	log := ctrllog.FromContext(ctx)
 	log.Info("Running StorageClassClaim controller in Consumer Mode")
+
+	providerClient, err := providerclient.NewProviderClient(ctx, storagecluster.Spec.ExternalStorage.StorageProviderEndpoint, time.Minute) // TODO what should be the timeout limit for synchronous operations on the client?
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// storageClass must have the same name as the storageClassClaim.
+	storageClass := &storagev1.StorageClass{}
+	storageClass.Name = r.storageClassClaim.Name
+
+	if r.storageClassClaim.GetDeletionTimestamp().IsZero() {
+
+		// Create the StorageClass (StorageClass.Name == StorageClassClaim.Name) with StorageClassClaim as its owner.
+		_, err = controllerutil.CreateOrUpdate(ctx, r.Client, storageClass, func() error {
+
+			// Update the StorageClassClaim status.
+			r.updateStatus(ctx, r.storageClassClaim, ocsv1alpha1.StorageClassClaimPending)
```

Comment:
We do not call `update` on the status of the storage class claim we are working on during the reconcile it self,  the status will always be saved at the end of the reconcile even if we exit with an error
Mainly because we want to ensure a single update on the status on each reconcile iteration. 

Also, pending does not mean much, maybe `configuring` is a better name

```suggestion
			r..storageClassClaim.Status.Phase = StorageClassClaimPending
```